pkg/app/piped/executor/analysis/analysis.go

```diff
@@ -96,6 +96,30 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 
 	eg, ctxWithTimeout := errgroup.WithContext(ctxWithTimeout)
 
+	// Sync the skip command.
+	var (
+		status = model.StageStatus_STAGE_NOT_STARTED_YET
+		doneCh = make(chan struct{})
+	)
+	defer close(doneCh)
+	go func() {
+		ticker := time.NewTicker(5 * time.Second)
+		defer ticker.Stop()
```

Comment:
If all stages of the analysis have already been done, this should be stopped gracefully too.